compiler/rustc_traits/src/chalk/lowering.rs

```diff
@@ -54,13 +54,13 @@ impl<'tcx> LowerInto<'tcx, chalk_ir::Substitution<RustInterner<'tcx>>> for Subst
         self,
         interner: &RustInterner<'tcx>,
     ) -> chalk_ir::Substitution<RustInterner<'tcx>> {
-        chalk_ir::Substitution::from_iter(interner, self.iter().map(|s| s.lower_into(interner)))
+        chalk_ir::Substitution::from_iter(*interner, self.iter().map(|s| s.lower_into(interner)))
```

Comment:
Should change the trait to take `interner` by value.

Reply:
Ah yes. I wondered about this. I'll try that out. 👍